icelake/tests/delta_test.rs

```diff
@@ -0,0 +1,421 @@
+            "Real path of table is: {}",
+            table.current_table_metadata().location
+        );
+        let mut delta_writer = table
+            .writer_builder()
+            .await
+            .unwrap()
+            .build_equality_delta_writer(vec![1, 2])
+            .await
+            .unwrap();
+
+        self.write_and_delete_with_delta(
+            &table,
+            &mut delta_writer,
+            Some(vec![
+                Arc::new(Int64Array::from(vec![1, 1, 1])),
+                Arc::new(Int64Array::from(vec![1, 2, 3])),
+                Arc::new(Int64Array::from(vec![1, 4, 5])),
+            ]),
+            None,
+        )
+        .await;
+
+        self.write_and_delete_with_delta(
+            &table,
+            &mut delta_writer,
+            Some(vec![
+                Arc::new(Int64Array::from(vec![1, 1, 1])),
+                Arc::new(Int64Array::from(vec![2, 3, 4])),
+                Arc::new(Int64Array::from(vec![2, 3, 4])),
+            ]),
+            None,
+        )
+        .await;
+
+        self.commit_writer(&mut table, delta_writer).await;
+
+        self.poetry.run_file(
+            "check.py",
+            [
+                "-s",
+                &self.spark_connect_url(),
+                "-q1",
+                "select * from s1.t1",
+                "-q2",
+                "select * from s1.t2",
+            ],
+            "Check select * from s1.t1",
+        )
+    }
+
+    pub async fn test_delete(&mut self) {
+        self.poetry.run_file(
+            "execute.py",
+            [
+                "-s",
+                &self.spark_connect_url(),
+                "-q",
+                "insert into s1.t2 values (1,1,1),(1,2,2),(1,3,3)",
+            ],
+            "insert into s1.t2 values (1,1,1),(1,2,2),(1,3,3)",
+        );
+
+        let mut table = self.create_icelake_table().await;
+        log::info!(
+            "Real path of table is: {}",
+            table.current_table_metadata().location
+        );
+        let mut delta_writer = table
+            .writer_builder()
+            .await
+            .unwrap()
+            .build_equality_delta_writer(vec![1, 2])
+            .await
+            .unwrap();
+
+        self.write_and_delete_with_delta(
+            &table,
+            &mut delta_writer,
+            Some(vec![
+                Arc::new(Int64Array::from(vec![1, 1, 1])),
+                Arc::new(Int64Array::from(vec![1, 2, 4])),
+                Arc::new(Int64Array::from(vec![1, 2, 4])),
+            ]),
+            None,
+        )
+        .await;
+
+        self.commit_writer(&mut table, delta_writer).await;
+
+        let mut delta_writer = table
+            .writer_builder()
+            .await
+            .unwrap()
+            .build_equality_delta_writer(vec![1, 2])
+            .await
+            .unwrap();
+
+        self.write_and_delete_with_delta(
+            &table,
+            &mut delta_writer,
+            Some(vec![
```

Comment:
I don't quite understand why the results are same? I think the result should be:
```
1 1 1
1 2 4
1 2 4
```

Reply:
We first insert 
```
row: 
1 1 1 
1 2 2 
1 4 4 
```
and then insert 
```
row: 
1 3 3 
1 5 5
```
and remove
```
row: 
1 4 4 
1 5 5
```